### PR TITLE
Fix the transformation of plane's normal

### LINKED ISSUE #9109: Transforming a plane with a matrix doesn't produce correct normal
When transforming a plane with a matrix, [the plane's normal is currently multiplied with the transform matrix itself](https://github.com/CesiumGS/cesium/blob/fa2edd6d0f77d2977e957a9334ef42a880a83897/Source/Core/Plane.js#L210). However, the transform matrix should be converted to normal matrix before multiplying with normal. This is the [sandcastle's example(Notice the angle printed out on the console)](https://sandcastle.cesium.com/#c=7VZNj9sgEP0rKCdH8uJE3Z42WXWba7u70lZ78oXYOIuKIQKcbFr1v3eA4K84cSq1t/pgGZiZ92bmAd4RhXaM7qlCSyToHq2oZlWJX91clE4yN15JYQgTVKWT6V0qUlFUIjNMClSwd5p/U0ToQqrymRNBo619x8iE2RgpqitupigVP1OB0A5QBSwQ/pUYxd4B+4jrx7fY+W6lplFvgYkdVTDdCt6ifbSKptMz03cBvvZ/dDxOGZRAmG354fPhWTJhHvQrzYxUUZt3jFyq2M91IFdEGfgi4kNAPVk4urEfrWwej5F6E77oPebPUjPXg+VA7Ib+S0Y4UVGX6Y0f5kwbIjJ6HfUzpWn34oTcwFRIRlFTKRGCO+ngQsnSBfV5R5fihVodxQVRf9nASYIyRYmhSCq2YQKa65L1K1IUTJWICGneqHJQiGkEVYRxzwOjR2lYRsF6w6k1m6Wi0e5g3ZueDld0jmcxmtkXfIFKbSGufC60yJJylLu72G/H0HNADbY+z65xK+asJtny6ZTsnGeT3s28cc1ZUQxWS1draGdmIk8o7oDEYxn7tgzFzaWJLGh3g1o/EICWnGIuN5Hz92oEadS6GlAC2jPzBm0XN5VgzkbDrqKodDvCudcagQzQDIYHVEpFPVNnfe6ss5q3m/ScZD7act46ycycZs6dbB7LdA7jBs5vsHo1nNItavGAdka67AnZd2hKDfAUNDZyenQIBB2MdL4GeeiK8k+grpbauHp7OcfD/C6CXKPlbmfHRe10CRd0S9rNSdjS696eayeyuXCz/03N/HO9/NfKuFYm8WShzYHTe3sffWLlViqDKsUjjBNDS+g7QCTrKvtODc60tk6LJLgscrZDLF8O/C6ijBOtYaWoOH+BezGd3C8SsO+4cUlyJjZP8G/HyQFMLI3F2/z+i1/AGC8SGFrQvq+Rkq+JasX9DQ)

This also happens in the `czm_transformPlane()` glsl function

Another explanation from stack overflow:
https://stackoverflow.com/questions/7685495/transforming-a-3d-plane-using-a-4x4-matrix

### PR DESCRIPTION
**Pull request by baothientran**
Fixes #9109 

This PR will convert the transformation matrix to a normal matrix before multiplying with the plane's normal. Most of the fix happens in glsl `czm_transformPlane()` function and `Cesium.Plane.transform()`

@lilleyse Can you please take a look at it?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @baothientran!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/plane-transform/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by baothientran** [comment]
It is weird that `prettier-check` on CI is failing some of the files that I don't modify

File: Source/Scene/Model.js [hunk 1]
```diff
@@ -4799,13 +4818,15 @@ function modifyShaderForClippingPlanes(
   shader +=
     "uniform highp sampler2D gltf_clippingPlanes; \n" +
     "uniform mat4 gltf_clippingPlanesMatrix; \n" +
+    "uniform mat4 gltf_normalClippingPlanesMatrix; \n" +
```
Comment (lilleyse) [inline Source/Scene/Model.js:4821]: The normal matrix should probably be a `Matrix3` everywhere. This keeps it consistent with `czm_normal` which is also a `Matrix3`.

**Comment by lilleyse** [comment]
@baothientran only one suggestion here (which is small but affects a lot of files). I tested non-uniform models and point clouds and it works well.

**Comment by baothientran** [comment]
@IanLilleyT it is indeed a lot simpler to do `transform * clippingPlane` with the `transform = inverse(transpose(originalMatrix))`. I will update it right away

**Comment by baothientran** [comment]
@IanLilleyT there is a little complication. It seems like `gltf_clippingPlanesMatrix` is also used to calculate IBL in this [code](https://github.com/CesiumGS/cesium/blob/c9630811eb3bdcdb69e633e7d6697e5fae0dd0b7/Source/Scene/processPbrMaterials.js#L1040). I'm not really sure about how IBL works. The PBR material behavior may be changed if I transpose and inverse the clipping plane matrix.

**Comment by baothientran** [comment]
@IanLilleyT I close this and open another PR #9135 instead. The approaches between two PR are quite difference, so it may be better to start a new one to prevent accidental changes

**Pull request closed without merging**